### PR TITLE
fix: db installation

### PR DESCRIPTION
**Pull request by blcksrx**
### SUMMARY

It's not a really good idea to pass the DB installation to the user without specifying a goo version, Also the best practice is providing the extras on the `extras_require`
In addition this PR, depends on this PR:
https://github.com/apache/incubator-superset/pull/10407
Because there's no way to pass the git link to the extras_require

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by blcksrx** [comment]
Actually I did fix the `black` issue! It would be nice another GitHub workflow for canceling the current job in case of new commit/ force push

File: docs/installation.rst [hunk 1]
```diff
@@ -482,72 +482,67 @@ connect to the databases you want to access through Superset.
...
+| Amazon Athena    | ``pip install "apache-superset[athena]"``                         | ``awsathena+jdbc://``                           |
++------------------+-------------------------------------------------------------------+-------------------------------------------------+
+| Amazon Redshift  | ``pip install  "apache-superset[redshift]"``                      | ``redshift+psycopg2://``                        |
++------------------+-------------------------------------------------------------------+-------------------------------------------------+
+| Apache Drill     | ``pip install "apache-superset[drill]"``                          | For the REST API:``                             |
+|                  |                                                                   | ``drill+sadrill://``                            |
+|                  |                                                                   | For JDBC                                        |
+|                  |                                                                   | ``drill+jdbc://``                               |
```
Comment (villebro) [inline docs/installation.rst:495]: As long as drill doesn't have an `extras_require` we probably need to state here that there is no PyPI package for drill yet, and just link to the GitHub repo.

File: docs/installation.rst [hunk 2]
```diff
@@ -482,72 +482,67 @@ connect to the databases you want to access through Superset.
...
-+------------------+---------------------------------------+-------------------------------------------------+
-| Vertica          | ``pip install                         |  ``vertica+vertica_python://``                  |
-|                  | sqlalchemy-vertica-python``           |                                                 |
-+------------------+---------------------------------------+-------------------------------------------------+
-| Hana             | ``pip install hdbcli sqlalchemy-hana``|  ``hana://``                                    |
-|                  | or                                    |                                                 |
-|                  | ``pip install apache-superset[hana]`` |                                                 |
-+------------------+---------------------------------------+-------------------------------------------------+
++------------------+-------------------------------------------------------------------+-------------------------------------------------+
+| database         | pypi package                                                      | SQLAlchemy URI prefix                           |
```
Comment (villebro) [inline docs/installation.rst:486]: Let's make the headers uniform:
```suggestion
| Database         | PyPI package                                                      | SQLAlchemy URI prefix                           |
```

**Review by villebro — CHANGES_REQUESTED** [review]
Looks good, a few suggestions!

File: docs/installation.rst [hunk 3]
```diff
@@ -482,72 +482,67 @@ connect to the databases you want to access through Superset.
...
+| Amazon Redshift  | ``pip install  "apache-superset[redshift]"``                      | ``redshift+psycopg2://``                        |
++------------------+-------------------------------------------------------------------+-------------------------------------------------+
+| Apache Drill     | ``pip install "apache-superset[drill]"``                          | For the REST API:``                             |
+|                  |                                                                   | ``drill+sadrill://``                            |
+|                  |                                                                   | For JDBC                                        |
+|                  |                                                                   | ``drill+jdbc://``                               |
++------------------+-------------------------------------------------------------------+-------------------------------------------------+
+| Apache Druid     | ``pip install "apache-superset[druid]"``                          | ``druid://``                                    |
```
Comment (john-bodley) [inline docs/installation.rst:497]: @villebro what are your thoughts about removing the `pip install` prefix here and just mentioning this as a primer, i.e., to install the PyPI package run `pip install <package>`.

File: setup.py [hunk 4]
```diff
@@ -120,11 +120,23 @@ def get_git_sha():
         "elasticsearch": ["elasticsearch-dbapi>=0.1.0, <0.2.0"],
         "druid": ["pydruid>=0.6.1,<0.7"],
         "hana": ["hdbcli==2.4.162", "sqlalchemy_hana==0.4.0"],
-        "dremio": ["sqlalchemy_dremio>=1.1.0"],
-        "cockroachdb": ["cockroachdb==0.3.3"],
+        "dremio": ["sqlalchemy-dremio>=1.1.5, <1.2"],
```
Comment (john-bodley) [inline setup.py:123]: Nit @blcksrx would you mind sorting these in alphabetical order? It's easier to grok which engines are supported using this approach.

File: docs/installation.rst (hunk 3 above)
Comment (villebro) [reply docs/installation.rst:497]: I agree @john-bodley , currently it is overly repetitive.

File: setup.py [hunk 5]
```diff
@@ -109,22 +109,36 @@ def get_git_sha():
...
-        "presto": ["pyhive[presto]>=0.4.0"],
-        "elasticsearch": ["elasticsearch-dbapi>=0.1.0, <0.2.0"],
-        "druid": ["pydruid>=0.6.1,<0.7"],
-        "hana": ["hdbcli==2.4.162", "sqlalchemy_hana==0.4.0"],
-        "dremio": ["sqlalchemy_dremio>=1.1.0"],
-        "cockroachdb": ["cockroachdb==0.3.3"],
-        "thumbnails": ["Pillow>=7.0.0, <8.0.0"],
-        "excel": ["xlrd>=1.2.0, <1.3"],
-        "prophet": ["fbprophet>=0.6, <0.7"],
+        {
```
Comment (john-bodley) [inline setup.py:112]: This should be a `Dict[str, List[str]` however it’s wrapped in an additional set of `{}` which is causing the CI issues.

**Review by john-bodley — APPROVED** [review]
Thanks @blcksrx this LGTM.

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**